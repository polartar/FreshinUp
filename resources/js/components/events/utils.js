import get from 'lodash/get'

const getFileNameCopy = (name) => {
  const regex = /\s*\(([0-9]+)\)$/gm
  const matches = name.match(regex) || []
  const count = (
    parseInt(
      get(matches, '[0]', '')
        .replace('(', '')
        .replace(')', '')
    ) || 0
  ) + 1
  return `Copy of ${name.replace(get(matches, '[0]', ''), '')} (${count})`
}

export default getFileNameCopy
