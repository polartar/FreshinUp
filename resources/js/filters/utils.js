export function prefixStr (value, prefix = '@') {
  if (!value) return ''
  return value ? `${prefix} ${value}` : ''
}
