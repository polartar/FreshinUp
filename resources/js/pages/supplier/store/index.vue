
<script>
import stores from '~/pages/admin/fleet-members/index.vue'
import { mapGetters } from 'vuex'
const INCLUDE = [
  'tags',
  'addresses',
  'status',
  'owner',
  'type'
]

export default {
  extends: stores,
  computed: {
    ...mapGetters(['currentUser'])
  },
  beforeRouteEnterOrUpdate (vm, to, from, next) {
    vm.setPageLoading(true)
    Promise.all([
      vm.$store.dispatch('stores/setFilters', {
        ...vm.$route.query,
        ...this.lastFilterParams,
        owner_uuid: vm.currentUser.uuid
      }),
      vm.$store.dispatch('storeStatuses/getItems', {
        params: { include: INCLUDE }
      })
    ])
      .then(() => {
        if (next) next()
      })
      .catch((error) => console.error(error))
      .then(() => vm.setPageLoading(false))
  }
}
</script>
