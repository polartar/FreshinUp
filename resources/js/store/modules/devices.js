import makeRestStore from 'fresh-bus/store/utils/makeRestStore'

export default (initialState = {}) => {
  const store = makeRestStore(
    'devices',
    { item: initialState.item, items: initialState.items },
    {
      itemsPath: () => `/foodfleet/devices`,
      itemPath: ({ id }) => `/foodfleet/devices/${id}`
    }
  )

  return {
    namespaced: true,
    ...store
  }
}
