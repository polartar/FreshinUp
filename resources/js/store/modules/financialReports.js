import makeRestStore from '@freshinup/core-ui/src/store/utils/makeRestStore'

export default (initialState = {}) => {
  const store = makeRestStore(
    'financialreports',
    { item: initialState.item, items: initialState.items },
    {
      itemsPath: () => `/foodfleet/financial-reports`,
      itemPath: ({ id }) => `/foodfleet/financial-reports/${id}`
    }
  )

  return {
    namespaced: true,
    ...store
  }
}
