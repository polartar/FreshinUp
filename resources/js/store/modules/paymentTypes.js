import makeRestStore from 'fresh-bus/store/utils/makeRestStore'

export default (initialState = {}) => {
  const store = makeRestStore(
    'paymenttypes',
    { item: initialState.item, items: initialState.items },
    {
      itemsPath: () => `/foodfleet/payment-types`,
      itemPath: ({ id }) => `/foodfleet/payment-types/${id}`
    }
  )

  return {
    namespaced: true,
    ...store
  }
}
