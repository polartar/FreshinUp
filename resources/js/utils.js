import get from 'lodash/get'
/**
 *
 * @param options HTML attributes
 * @param [options.type] string 'link'|'script' or anything in fact
 * @param [options.container] HTMLElement
 * @returns {Promise<unknown>}
 */
export const lazyLoad = (options) => {
  return new Promise((resolve, reject) => {
    if (!options.type) {
      return reject(new Error('options.type is required'))
    }
    if (!options.container) {
      options.container = document.body
    }
    const { type, container, ...attributes } = options
    let element = document.createElement(type)
    Object.keys(attributes).forEach(key => element.setAttribute(key, attributes[key]))
    element.addEventListener('load', () => {
      resolve(element)
    })
    element.addEventListener('error', (e) => {
      reject(e)
    })
    container.append(element)
  })
}

export const getFileNameCopy = (name) => {
  const regex = /\s*\(([0-9]+)\)$/gm
  const matches = name.match(regex) || []
  const count = (
    parseInt(
      get(matches, '[0]', '')
        .replace('(', '')
        .replace(')', '')
    ) || 0
  ) + 1
  return `Copy of ${name.replace(get(matches, '[0]', ''), '')} (${count})`
}
