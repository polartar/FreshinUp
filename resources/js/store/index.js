import { createStore } from 'fresh-bus/store'

import page from './modules/page'
import devices from './modules/devices'
import financialModifiers from './modules/financialModifiers'
import financialReports from './modules/financialReports'
import paymentTypes from './modules/paymentTypes'
import financialsummary from './modules/financialsummary'
import squares from './modules/squares'
import documents from './modules/documents'
import documentStatuses from './modules/documentStatuses'
import documentTypes from './modules/documentTypes'
import transactions from './modules/transactions'
import companyDetails from './modules/companyDetails'
import eventSummary from './modules/eventSummary'
import eventTypes from './modules/eventTypes'
import eventHistories from './modules/eventHistories'
import venues from './modules/venues'
import stores from './modules/stores'
import storeStatuses from './modules/storeStatuses'
import storeTypes from './modules/storeTypes'
import companies from './modules/companies'

export default (initialState = {}) => {
  return createStore(
    initialState,
    {
      modules: {
        page: page({}),
        devices: devices({}),
        financialModifiers: financialModifiers({}),
        financialReports: financialReports({}),
        paymentTypes: paymentTypes({}),
        financialsummary: financialsummary({}),
        squares: squares({}),
        documents: documents({}),
        documentStatuses: documentStatuses({}),
        documentTypes: documentTypes({}),
        transactions: transactions({}),
        companyDetails: companyDetails({}),
        eventSummary: eventSummary({}),
        eventTypes: eventTypes({}),
        eventHistories: eventHistories({}),
        venues: venues({}),
        stores: stores({}),
        storeStatuses: storeStatuses({}),
        storeTypes: storeTypes({}),
        companies: companies({})
      }
    }
  )
}
