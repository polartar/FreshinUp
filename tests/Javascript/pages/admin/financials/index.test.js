import { mount } from '@vue/test-utils'
import { createStore } from 'fresh-bus/store'
import { createLocalVue } from 'fresh-bus/tests/utils'
import { FIXTURE_REPORTABLES, FIXTURE_REPORTABLES_RESPONSE } from 'tests/__data__/reportables'
import Component from '~/pages/admin/financials/index.vue'
import devices from '~/store/modules/devices'
import paymentTypes from '~/store/modules/paymentTypes'
import financialReports from '~/store/modules/financialReports'

describe('Admin Financial Reports Page', () => {
  let localVue, mock
  describe('Mount', () => {
    beforeEach(() => {
      const vue = createLocalVue({ validation: true })
      localVue = vue.localVue
      mock = vue.mock
    })
    afterEach(() => {
      mock.restore()
    })
    test('snapshot', done => {
      const vue = createLocalVue({ validation: true })
      localVue = vue.localVue
      mock = vue.mock
        .onGet('api/foodfleet/financial-reports', { params: { 'page[size]': 10, 'page[number]': 1 } })
        .reply(200, { data: FIXTURE_REPORTABLES })
        .onGet('api/foodfleet/financial-reports').reply(200, FIXTURE_REPORTABLES_RESPONSE)

      mock.onGet('api/foodfleet/devices')
        .reply(200, {})

      mock.onGet('api/foodfleet/payment-types')
        .reply(200, {})

      mock.onAny().reply(config => {
        console.warn('No mock match for ' + config.url, config)
        return [404, { message: 'No mock match for ' + config.url, data: config }]
      })

      const store = createStore({
        financialReports: {
          items: FIXTURE_REPORTABLES_RESPONSE
        }
      }, {
        modules: {
          devices: devices({}),
          paymentTypes: paymentTypes({}),
          financialReports: financialReports({})
        }
      })

      const wrapper = mount(Component, {
        localVue: localVue,
        store
      })

      // Action: load the page data
      Component.beforeRouteEnterOrUpdate(wrapper.vm, null, null, async () => {
        await wrapper.vm.$nextTick()
        expect(wrapper.element).toMatchSnapshot()
        done()
      })
    })
  })
})
