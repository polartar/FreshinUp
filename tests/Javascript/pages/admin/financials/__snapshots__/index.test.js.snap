// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Financial Reports Page Mount snapshot 1`] = `
<div>
  <v-dialog-stub
    contentclass="right-dialog"
    maxwidth="436"
    origin="center center"
    scrollable="true"
    transition="dialog-transition"
    width="auto"
  >
    <advanced-filter-stub
      advancedfilters="[object Object]"
      basefilters="[object Object]"
      devices=""
      maxwidthbottomrow="422"
      paymenttypes=""
    />
  </v-dialog-stub>
   
  <v-layout-stub
    row=""
    tag="div"
    wrap=""
  >
    <v-flex-stub
      align-center=""
      d-inline-flex=""
      ma-2=""
      tag="div"
    >
      <h2>
        Financial Reports
      </h2>
    </v-flex-stub>
     
    <v-flex-stub
      tag="div"
      text-xs-right=""
    >
      <v-btn-stub
        activeclass="v-btn--active"
        color="primary"
        dark="true"
        tag="button"
        type="button"
      >
        
        Advanced Filters
      
      </v-btn-stub>
    </v-flex-stub>
  </v-layout-stub>
   
  <v-divider-stub />
   
  <br />
   
  <v-card-stub
    class="pa-4"
    tag="div"
  >
    <basic-filter-stub
      filters="[object Object]"
    />
  </v-card-stub>
   
  <v-layout-stub
    px-5=""
    py-2=""
    row=""
    tag="div"
  >
    <v-flex-stub
      tag="div"
    >
      <v-card-stub
        tag="div"
      >
        <v-card-title-stub>
          <p
            class="font-weight-bold mt-3"
          >
            
            Your Saved Financial Reports
          
          </p>
        </v-card-title-stub>
         
        <reportable-filter-stub />
         
        <reportable-list-stub
          baseurl="/admin/transactions"
          must-sort=""
          page="1"
          reportables="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
          rowsperpage="10"
          rowsperpageitems="5,10,15,25,30,50"
          selectables="[object Object]"
          sortby=""
          totalitems="0"
        />
      </v-card-stub>
    </v-flex-stub>
  </v-layout-stub>
   
  <v-dialog-stub
    contentclass=""
    maxwidth="500"
    origin="center center"
    transition="dialog-transition"
    width="auto"
  >
    <simple-confirm-stub
      cancellabel="No"
      class=""
      oklabel="Yes"
      title="Are you sure you want to delete this report?"
    >
      <div
        class="py-5 px-2"
      >
        <p
          class="subheading"
        >
          <span>
            Report
          </span>
          
            : 
          
        </p>
      </div>
    </simple-confirm-stub>
  </v-dialog-stub>
</div>
`;
